packages/core/crates/core-crypto/README.md

```diff
@@ -3,4 +3,39 @@
 This Rust crate contains cryptographic primitives and basic types that
 are used in the HOPR protocol.
 
-The higher-level operations are meant to be in separate crates, like `core-packet`.
+The higher-level operations are meant to be in the separate crates, like `core-packet`.
+
+The crate is structured into the following modules:
+
+- `derivation`: contains simple key derivation functions for different purposes
+- `ec_groups`: implementation of `SphinxSuite` trait, see the chapter below.
+- `error`: various errors produced by this crate
+- `iterated_hash`: hash commitment implementation for the tickets
+- `keypairs`: defines the two types of keypairs (`ChainKeypair` used for on-chain operations - based on secp256k1, and `OffchainKeypair` for packet operations - based on Ed25519)
+- `parameters`: contains various cryptography related global constants
+- `prg`: implementation of a pseudo-random generator function used in SPHINX packet header construction
+- `primitives`: contains implementation of cryptographic primitives: Blake2s256 digest, Keccak256 digest, Mac using Blake2s256 and ChaCha20 stream cipher
+- `prp`: implementation of the Lioness wide-block cipher using Chacha20 and Blake2b256
+- `random`: all functions that generate something requiring randomness source are implemented here
+- `routing`: implements the SPHINX header
+- `shared_keys`: derivation of shared keys for SPHINX header (see below)
+- `types`: general use types for the entire code base, e.g. public keys, signatures,...
+- `utils`: generic utility types and functions used by this crate
```

Comment:
We should start considering splitting this package into more components.

Reply:
`routing` and `shared_keys` might probably go into smth like `core-sphinx` in the future.